frontend/lib/src/components/widgets/DataFrame/hooks/useTableSizer.ts

```diff
@@ -100,19 +107,20 @@ function useTableSizer(
     }
   }
 
-  let initialWidth: number | undefined // If container width is undefined, auto set based on column widths
-  let maxWidth = containerWidth
+  let initialWidth: number | undefined
+  let maxWidth = availableWidth
 
   if (element.useContainerWidth) {
-    // Always use the full container width
-    initialWidth = containerWidth
+    // If user has set use_container_width,
+    // use the full container width.
+    initialWidth = availableWidth
   } else if (element.width) {
     // User has explicitly configured a width
     initialWidth = Math.min(
       Math.max(element.width, MIN_TABLE_WIDTH),
-      containerWidth
+      availableWidth
     )
-    maxWidth = Math.min(Math.max(element.width, maxWidth), containerWidth)
+    maxWidth = Math.min(Math.max(element.width, maxWidth), availableWidth)
   }
```

Comment:
I think the multiple usage of nested min-max calls make it sufficiently hard to read and understand to warrant moving this into its own function and add some comments 😅 I think it would be good to have a list of the different widths in bullet points below each other giving a brief explanation why its needed and what the differences are and how it is calculated in a few words.

Reply:
I added a lot more comments 👍